### PR TITLE
feat(tests): Restructured and Expanded Integration Tests for CinderGraph

### LINKED ISSUE #118: feat(tests): Restructure and Expand Integration Tests for CinderGraph
## Summary
The current integration tests for CinderGraph reside in a single file located at ``tests/integration/CinderGraph/CinderGraphShard.cpp`` are minimal and only cover basic operations like addVertex and addEdge.
To improve coverage, clarity, and maintainability, we propose restructuring the integration test suite into a modular format, separating per-method functional tests and scenario-based tests. 

This issue defines the structure and provides guidance for a starting PR.

## Proposed directory structure
```sh
tests/
└── integration/
    └── CinderGraph/
        ├── functional/              # Per-method integration tests
        │   ├── addVertex_test.cpp
        │   ├── removeVertex_test.cpp
        │   ├── addEdge_test.cpp       // weighted & unweighted edges
        │   ├── removeEdge_test.cpp
        │   ├── updateEdge_test.cpp
        │   ├── getEdge_test.cpp
        │   ├── clearEdges_test.cpp
        │   ├── hasVertex_test.cpp
        │   └── addEdges_test.cpp
        │
        ├── scenarios/               # Multi-step integration workflows
        │   ├── BasicConnectivity_test.cpp
        │   ├── UpdateAndQueryFlow_test.cpp
        │   ├── BulkInsertionStress_test.cpp
        │   ├── ClearAndReuseGraph_test.cpp
        │   └── PolicyInteraction_test.cpp
        │
        └── common/                  # Shared test utilities
            ├── DummyGraphBuilder.hpp
            └── TestUtils.hpp
```
## 1. functional/ – Per-method tests

Each file should:
- Focus on one public API function.
- Include normal use cases and edge cases.
- Reuse the dummy graph from ``DummyGraphBuilder`` to create a predictable baseline.

Examples of expected content:
- ``addVertex_test.cpp``: add new vertices, check existence, test duplicate insertion.
- ``removeVertex_test.cpp``: remove existing/non-existing vertices, check adjacency updates.
- ``addEdge_test.cpp``: test both weighted & unweighted edge insertion, verify adjacency and weight.
- ``removeEdge_test.cpp``: remove edges, check connectivity remains correct.
- ``updateEdge_test.cpp``: update weights, test invalid updates.
- ``getEdge_test.cpp``: validate edge retrieval correctness, check exceptions.
- ``clearEdges_test.cpp``: verify clearing edges does not remove vertices.
- ``hasVertex_test.cpp``: check vertex existence queries.

## 2. scenarios/ – Multi-step integration workflows
These files simulate realistic sequences of operations to validate interactions between multiple functions. For the starting PR, contributors can add 3-4 basic scenario tests:

- ``BasicConnectivity_test.cpp``: add vertices + edges, remove edges, check connectivity remains consistent.
- ``UpdateAndQueryFlow_test.cpp``: add/update edges, query weights, verify correctness of cumulative operations.
- ``ClearAndReuseGraph_test.cpp``: clear edges, reuse graph for new insertions, verify no residual data.
(Optional for starting PR: BulkInsertionStress_test.cpp & PolicyInteraction_test.cpp can be added later.)
- ``BulkInsertionStress_test.cpp``: Bulk edge addition and vertex additions, etc, weighted and unweighted all cases.

##  3. common/ – Shared test utilities
- ``DummyGraphBuilder.hpp``: provides helper functions to create small, predictable graphs (directed/undirected, weighted/unweighted).
- ``TestUtils.hpp``: assertion helpers, edge/vertex comparison utilities, logging helpers for debugging.

Tests are integrated into CI (CMake/GoogleTest) and pass.

### PR DESCRIPTION
**Pull request by me-hem**
This PR restructures and expands the CinderGraph integration test suite from a single monolithic test file into a modular, well-organized structure that improves maintainability, clarity, and test coverage (issue #118). The previous integration tests (`tests/integration/CinderGraph/CinderGraphShard.cpp`) were minimal and covered only basic operations. The new structure introduces dedicated files for each graph operation and scenario, allowing for more comprehensive testing.

## New Structure

```
tests/integration/CinderGraph/
├── common/                          # Shared test utilities
│   └── DummyGraphBuilder.hpp       # Graph factory methods
│
├── functional/                      # Per-method integration tests
│   ├── addVertex_test.cpp          # Vertex addition tests
│   ├── removeVertex_test.cpp       # Vertex removal tests
│   ├── addEdge_test.cpp            # Edge addition (weighted/unweighted)
│   ├── removeEdge_test.cpp         # Edge removal tests
│   ├── updateEdge_test.cpp         # Edge weight update tests
│   ├── getEdge_test.cpp            # Edge retrieval tests
│   ├── clearEdges_test.cpp         # Edge clearing tests
│   └── hasVertex_test.cpp          # Vertex existence checks
│
└── scenarios/                       # Multi-step integration workflows
    ├── BasicConnectivity_test.cpp   # Basic graph operations flow
    ├── UpdateAndQueryFlow_test.cpp  # Update and query patterns
    ├── BulkInsertionStress_test.cpp # Stress testing with bulk ops
    └── ClearAndReuseGraph_test.cpp  # Graph clearing and reuse
```

### 1. Common Utilities (common/)

- `DummyGraphBuilder.hpp` - Provides factory methods for creating test graphs with consistent configurations

### 2. Functional Tests (functional/)

- `addVertex_test.cpp`: Vertex insertion, duplicate handling
- `removeVertex_test.cpp`: Vertex deletion, non-existent vertex handling
- `addEdge_test.cpp`: Edge creation for all type combinations
- `removeEdge_test.cpp`: Edge deletion, weight retrieval on removal
- `updateEdge_test.cpp`: Weight updates, failure cases
- `getEdge_test.cpp`: Edge retrieval, optional weight handling
- `clearEdges_test.cpp`: Edge clearing while preserving vertices
- `hasVertex_test.cpp`: Vertex existence queries

### 3. Scenario Tests (scenarios/)

- `BasicConnectivity_test.cpp`: 
  - Add vertices and edges
  - Remove edges and verify connectivity
  - Remove vertices and check cascade effects
  - Complex graph manipulations

- `UpdateAndQueryFlow_test.cpp`:
  - Add → Update → Query patterns
  - Multiple sequential updates
  - Mixed operations with repeated queries
  - Remove → Add → Update cycles

- `BulkInsertionStress_test.cpp`:
  - Bulk vertex insertion 
  - Bulk edge insertion 
  - Dense graph creation
  - Mixed bulk operations with updates

- `ClearAndReuseGraph_test.cpp`:
  - Clear and verify state
  - Reuse after clearing
  - Stress test with repeated clear cycles
  - Empty graph clearing

### CODE REVIEW

File: tests/integration/CinderGraph/scenerios/BasicConnectivity_test.cpp [hunk 1]
```diff
@@ -0,0 +1,119 @@
+#include "../common/DummyGraphBuilder.hpp"
```
Comment (SharonIV0x86) [inline tests/integration/CinderGraph/scenerios/BasicConnectivity_test.cpp]: The folder name needs to be corrected.

Comment (me-hem) [reply tests/integration/CinderGraph/scenerios/BasicConnectivity_test.cpp]: Changes pushed.

**Comment by me-hem** [comment]
@SharonIV0x86 I have updated all references to the DummyGraphBuilder header file and modified CMakeLists.txt to build this header file too. You can now consider it for review.

File: tests/integration/CinderGraph/scenerios/BasicConnectivity_test.cpp (hunk 1 above)
Comment (SharonIV0x86) [reply tests/integration/CinderGraph/scenerios/BasicConnectivity_test.cpp]: @me-hem i believe it should be ``scenarios`` not ``scenerios``

Comment (me-hem) [reply tests/integration/CinderGraph/scenerios/BasicConnectivity_test.cpp]: Oops I missed that

Comment (me-hem) [reply tests/integration/CinderGraph/scenerios/BasicConnectivity_test.cpp]: @SharonIV0x86 Changes pushed.

**Review by SharonIV0x86 — APPROVED** [review]
Looks good to me.

**Comment by SharonIV0x86** [comment]
@me-hem Thank you for contributing your PR has been merged.
